docs/DeveloperGuide.md

```diff
@@ -2,12 +2,111 @@
+Below are the design and implementation details of KaChinnnng.
+
+### Design
+
+The architectural diagram of KaChinnnng is as follows:
+
+![ArchitectureDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/ArchitectureDiagram.png?raw=true)
+
+The kaChinnnng program will first enter the `run` state where the `storage` class will be accessed and data will be retrieved if the application has been used before.
+
+Next, the `User` will input various `commands` via the `Ui` class. The `commands` will then be parsed by the `Parser` class and the relevant `managers` will be called to execute the `commands`.
+
+Once the `User` inputs has been successfully executed, `commands` will make use of the `Ui` class to output the relevant information to the user.
+
+When the `User` exits the program, the `storage` class will be called to save the data to the local storage.
+
+`Storage` will read from the local storage when the program is run again.
+
+### Expense Class Overview
+
+The Expenses are divided into three categories. Transport, Utilities and Food.
+Each of the categories inherit from a base Expense class which in turn
+inherits from the FinancialRecord class.
+
+![expense_class_diagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/expense_class_diagram.png?raw=true)
+### IncomeManager
+
+the IncomeManager is facilitated by `IncomeParser`, `IncomeManager`, `IncomeList` and `Ui`.
+
+1. The user inputs the command to add income. This input is then passed to `IncomeParser` to be parsed.
+2. `IncomeParser` then parses the input and returns a `Income` object.
+3. `IncomeManager` then adds the `Income` object to the `IncomeList`.
+4. `IncomeManager` then returns a `String` to `Ui` to be printed out to the user.
+
+![IncomeManager_seqdiagram.drawio.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/IncomeManager_seqdiagram.drawio.png?raw=true)
 
+### ExpenseManager
+
+The ExpenseManager is facilitated by `ExpenseParser`,  `ExpenseManager`, `ExpenseList` and `Ui`
+
+1. The user inputs the command to add expense.
+2. This command is used to instantiate a ExpenseManager object which is executed.
+3. During execution, the command is passed to ExpenseParser which extracts and parses the input fields.
+4. Input field information is passed to the relevant Expense constructor which returns a Expense object.
+5. Expense object is returned to ExpenseManager from ExpenseParser.
+6. Expense obejct is returned to Duke from ExpenseManager.
+
+![expenseManager_sequence_diagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/expenseManager_sequence_diagram.png?raw=true)
+
+### Find Command
+The FindCommand is facilitated by `FindCommand`, `FindParser`, `Ui`,`ExpenseList` and `IncomeList`.
+
+1. The user will first input the command to find the expense or income.
+```
+find /t <type> /cat [category] /de [description] /date [date]
```

Comment:
It may be a little confusing as to why <type> is in <> while everything else is in []. This may be confusing to someone reading this, so maybe you can add a line explaining why it is different?